File: Arrays_&_Hashing/Longest_Consecutive_Sequence/2_Better.java
import java.util.*;

class Solution {
    public int longestConsecutive(int[] nums) {
        if (nums.length == 0) return 0;
        Arrays.sort(nums);
        int best = 1, cur = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i-1]) continue; // skip duplicates
            if (nums[i] == nums[i-1] + 1) {
                cur++;
            } else {
                best = Math.max(best, cur);
                cur = 1;
            }
        }
        best = Math.max(best, cur);
        return best;
    }
}

File: Arrays_&_Hashing/Longest_Consecutive_Sequence/2_Better.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(), nums.end());
        int best = 1, cur = 1;
        for (size_t i = 1; i < nums.size(); ++i) {
            if (nums[i] == nums[i-1]) continue; // skip duplicates
            if (nums[i] == nums[i-1] + 1) {
                ++cur;
            } else {
                best = max(best, cur);
                cur = 1;
            }
        }
        best = max(best, cur);
        return best;
    }
};
